### PR TITLE
manifest: Update hal nordic repo

### PR DESCRIPTION
**Pull request by JakubPegza**
Fix GPIOTE channel allocation

### CODE REVIEW

**Review by tejlmand — CHANGES_REQUESTED** [review]
This PR is targeted for v1.5-branch, so I think:
```
nrfconnect/sdk-hal_nordic@49b4055 (fix/GPIOTE-channel-allocation)
```

should be on a `v1.5-branch`, and not `fix/GPIOTE-channel-allocation`.

Ideally it should be a `vX.Y.Z-ncs-branch` but as `hal_nordic` doesn't have any release tags then I think `v1.5-branch` is most appropriate in this case.
@carlescufi thoughts on this ?

Also the commit:
https://github.com/nrfconnect/sdk-hal_nordic/commit/49b40553b9efe8d6fa3332b4ce67103f2bdcc087
should have `[nrf fromtree]` in it's title, as it is a cherry-pick from `hal_nordic/master`.

**Comment by tejlmand** [comment]
Please update manifest SHA to: [b96360f2446b772c2495ca49d4137b9b8f4bb2a8](https://github.com/nrfconnect/sdk-hal_nordic/commit/b96360f2446b772c2495ca49d4137b9b8f4bb2a8)

**Review by tejlmand — APPROVED** [review]
LGTM but would like @carlescufi to take a look also.

**Comment by carlescufi** [comment]
I wonder if `v1.5-branch` is the right branch name. We need to reconsider the way we tag branches, because it is really hard to follow. The alternative would be to use the Zephyr versioning scheme, which would be `v2.4.99-ncs1-branch`, since this is a project that doesn't have its own revision at all, to indicate that this is the hal_nordic that came with Zephyr 2.4.99 with a few additions. That said this is not that useful either, because we have modified hal_nordic enough for this not to be that meaningful.

**Comment by JakubPegza** [comment]
> I wonder if `v1.5-branch` is the right branch name. We need to reconsider the way we tag branches, because it is really hard to follow. The alternative would be to use the Zephyr versioning scheme, which would be `v2.4.99-ncs1-branch`, since this is a project that doesn't have its own revision at all, to indicate that this is the hal_nordic that came with Zephyr 2.4.99 with a few additions. That said this is not that useful either, because we have modified hal_nordic enough for this not to be that meaningful.

AFAIK no official branch naming scheme is present in sdk-hal_nordic, but I'm fine with both propositions, `v1.5` signifies that this branch was specifically created for NCS v1.5, going around hal_nordic as defined by zephyr, while `v2.4.99-ncs1-branch` would signify it was selected by this version of Zephyr before added fix. We're circumventing zephyr manifest with this fix, so I decided to go with ncs-branch name. If you think zephyr-branch name is more appropriate, this can be easily changed.

**Comment by carlescufi** [comment]
> > I wonder if `v1.5-branch` is the right branch name. We need to reconsider the way we tag branches, because it is really hard to follow. The alternative would be to use the Zephyr versioning scheme, which would be `v2.4.99-ncs1-branch`, since this is a project that doesn't have its own revision at all, to indicate that this is the hal_nordic that came with Zephyr 2.4.99 with a few additions. That said this is not that useful either, because we have modified hal_nordic enough for this not to be that meaningful.
> 
> AFAIK no official branch naming scheme is present in sdk-hal_nordic, but I'm fine with both propositions, `v1.5` signifies that this branch was specifically created for NCS v1.5, going around hal_nordic as defined by zephyr, while `v2.4.99-ncs1-branch` would signify it was selected by this version of Zephyr before added fix. We're circumventing zephyr manifest with this fix, so I decided to go with ncs-branch name. If you think zephyr-branch name is more appropriate, this can be easily changed.

I am a bit unsure about this. I am thinking that perhaps we should start using something like `ncs-v1.5.0` for this type of repo that has no upstream but at the same time it is a fork of an external repo (in this case hosted by zephyrproject-rtos). @tejlmand and @mbolivar-nordic do you have thoughts on this?

**Comment by mbolivar-nordic** [comment]
> this type of repo that has no upstream but at the same time it is a fork of an external repo (in this case hosted by zephyrproject-rtos)

This feels like a contradiction in terms. If it's a fork, it has an upstream, and our policy would force us to use upstream's tags and versions if any exist. But I think upstream zephyr doesn't tag or version its modules, right?

So in that case just coming up with our own numbering seems like the best we can do, and ncs-vX.Y.Z seems nice and readable. I've previously suggested we consider doing something like that for all of our repositories, since mapping a tag name to the version of NCS it was used in is currently difficult.

**Comment by carlescufi** [comment]
> > this type of repo that has no upstream but at the same time it is a fork of an external repo (in this case hosted by zephyrproject-rtos)
> 
> This feels like a contradiction in terms. If it's a fork, it has an upstream, and our policy would force us to use upstream's tags and versions if any exist. But I think upstream zephyr doesn't tag or version its modules, right?
> 
> So in that case just coming up with our own numbering seems like the best we can do, and ncs-vX.Y.Z seems nice and readable. I've previously suggested we consider doing something like that for all of our repositories, since mapping a tag name to the version of NCS it was used in is currently difficult.

I agree, let's go with `ncs-vX.Y.Z`, as we will likely anyway switch to this format for all non-core repos.

**Review by carlescufi — APPROVED** [review]
